crypto/evp/evp_lib.c

```diff
@@ -1283,9 +1281,9 @@ int EVP_CIPHER_CTX_get_algor_params(EVP_CIPHER_CTX *ctx, X509_ALGOR *alg)
      */
     type = alg->parameter;
 
-    derk = params[i].key;
-    derl = params[i].return_size;
-    if (i >= 0 && (der = OPENSSL_malloc(derl)) != NULL) {
+    if (i >= 0 && (der = OPENSSL_malloc(params[i].return_size)) != NULL) {
+        size_t derl = params[i].return_size;
+        const char *derk = params[i].key;
```

Comment:
It would be better to split the condition in 2 steps and just error out if `i < 0`.